### PR TITLE
User profile cleanup rayeka

### PR DESCRIPTION
**Pull request by armitage420**
## Name of User-Profile-Cleanup

[Notion Task Link](https://www.notion.so/User-Profiles-Cleanup-8208416ceae940cba463f8a43c7cdd5e)

#### Branches for testing

Frontend: `User-Profile-Cleanup-Rayeka`

Backend: `User-Profile-Cleanup`

### CODE REVIEW

File: src/components/Dashboard/AcademicDetails.js [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
 import icons from '../../utils/getIcons';
 import { AcademicDetailsModal } from './AcademicDetailsModal';
 import { Row } from './Row';
 
 export const AcademicDetails = ({ user }) => {
   const [modalShow, setModalShow] = useState(false);
-
+  const loginState = useSelector((state) => state.loginState);
+  const currentUsername = loginState.loggedIn ? loginState.user.username : null;
+  console.log(user);
```
Comment (patheticGeek) [inline src/components/Dashboard/AcademicDetails.js:12]: Remove it

File: src/pages/Profile.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+export default function Profile() {
+  const { username } = useParams();
+  const [user, setUser] = useState(null);
+  const [error, setError] = useState('');
+  const loginState = useSelector((state) => state.loginState);
+
+  let history = useHistory();
+  if (loginState.loggedIn && username === loginState.user.username) {
+    history.push('/');
+  }
```
Comment (patheticGeek) [inline src/pages/Profile.js:23]: Move inside useEffect

File: src/pages/Profile.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import { ProjectsComingSoon } from '../components/Dashboard/ProjectsComingSoon';
+import axios from '../config/axios.config';
+import { API_ENDPOINTS } from '../constants/api';
+import myLog from '../utils/myLog';
+
+export default function Profile() {
+  const { username } = useParams();
+  const [user, setUser] = useState(null);
+  const [error, setError] = useState('');
+  const loginState = useSelector((state) => state.loginState);
```
Comment (patheticGeek) [inline src/pages/Profile.js:18]: Change to useLoginState

File: src/components/Dashboard/BasicDetailsModal.js [hunk 4]
```diff
@@ -81,6 +82,28 @@ export const BasicDetailsModal = ({ modalProps, user, id }) => {
...
+            <input
+              type="text"
+              className="p-2 ml-3 w-100"
+              value={details.username}
+              onChange={(e) =>
+                setDetails({ ...details, username: e.target.value })
+              }
+              style={{
+                boxShadow: '0px 0px 10px #8264B433',
+                border: '0.800000011920929px solid #D6CCE6',
```
Comment (patheticGeek) [inline src/components/Dashboard/BasicDetailsModal.js:101]: There's no such thing as half a pixel change to 1px

**Review by patheticGeek — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/AcademicDetails.js (hunk 1 above)
Comment (armitage420) [reply src/components/Dashboard/AcademicDetails.js]: done

File: src/components/Dashboard/BasicDetailsModal.js (hunk 4 above)
Comment (armitage420) [reply src/components/Dashboard/BasicDetailsModal.js:102]: done

**Review by patheticGeek — APPROVED** [review]
LGTM

**Pull request merged**